packages/typespec-ts/src/utils/interfaces.ts

```diff
@@ -0,0 +1,14 @@
+import { RLCOptions } from "@azure-tools/rlc-common";
+import { SdkContext as TCGCSdkContext } from "@azure-tools/typespec-client-generator-core";
+
+export interface SdkContext extends TCGCSdkContext {
+  rlcOptions?: RLCOptions;
+  generationPathDetail?: GenerationDirDetail;
+}
+
+export interface GenerationDirDetail {
+  rootDir: string;
+  rlcSourcesDir: string;
+  modularSourcesDir?: string;
+  metadataDir: string;
```

Comment:
is rootDIr not equals to metadataDir ?

Reply:
Generally rootDir and metadataDir are the same, I proposed a new option because I will use rootDir to calculate if this is in sdk repo and I think using metadata one would be wired.